neonvm-daemon/cmd/main.go

```diff
@@ -26,24 +32,76 @@ func main() {
 	defer logger.Sync() //nolint:errcheck // what are we gonna do, log something about it?
 
 	logger.Info("Starting neonvm-daemon", zap.String("addr", *addr))
+	cpuHotPlugController := &cpuscaling.CPUSysFsStateScaler{}
+	srv := cpuServer{
+		cpuOperationsMutex:  &sync.Mutex{},
+		cpuSystemWideScaler: cpuHotPlugController,
+		logger:              logger.Named("cpu-srv"),
+	}
+	srv.run(*addr)
+}
 
-	srv := cpuServer{}
-	srv.run(logger, *addr)
+type cpuServer struct {
+	// Protects CPU operations from concurrent access to prevent multiple ensureOnlineCPUs calls from running concurrently
+	// and ensure that status response is always actual
+	cpuOperationsMutex  *sync.Mutex
+	cpuSystemWideScaler *cpuscaling.CPUSysFsStateScaler
+	logger              *zap.Logger
+}
+
+func (s *cpuServer) handleGetCPUStatus(w http.ResponseWriter) {
+	// should be replaced with cgroups milliseconds to be exposed instead of having CPU
+	s.cpuOperationsMutex.Lock()
+	defer s.cpuOperationsMutex.Unlock()
+	activeCPUs, err := s.cpuSystemWideScaler.GetActiveCPUsCount()
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+	if _, err := w.Write([]byte(fmt.Sprintf("%d", activeCPUs*1000))); err != nil {
+		s.logger.Error("could not write response", zap.Error(err))
+	}
 }
 
-type cpuServer struct{}
+func (s *cpuServer) handleSetCPUStatus(w http.ResponseWriter, r *http.Request) {
+	// TODO: should the call to this method be conditional, only if the statefs cpu scaling is enabled?
+	// on the other hand, currently this endpoint is called by runner only if the statefs scaling is enabled
+	// and it is a bit tricky to pass vmSpec here
```

Comment:
commenting on the TODO: it's not worth it to double-check that here. Let's trust what the runner says.